extension/intents/tabs/tabs.js

```diff
@@ -597,3 +598,166 @@ intentRunner.registerIntent({
     context.displayText(message);
   },
 });
+
+intentRunner.registerIntent({
+  name: "tabs.selectAllTabs",
+  async run(context) {
+    const activeTab = await context.activeTab();
+    const tabs = await browser.tabs.query({ currentWindow: true});
+    const allWindows = await browser.windows.getAll({
+      windowTypes: ["normal"],
+    });
+
+    const matchingTabs = await getMatchingTabs({
+      tabs,
+      activeTab,
+      sort_by_index: true,
+      allWindows: context.parameters.allWindows === "false"
+    });
+
+    if (matchingTabs.length === 0) {
+      const exc = new Error("No tab that matches the query");
+      exc.displayMessage = "There is no tab that matches the query";
+      throw exc;
+    }
+
+    await selectAllTabs(matchingTabs);
+
+  },
+});
+
+async function selectAllTabs(tabs){
+  if(tabs.length !== 0){
+    browser.tabs.query({
+      currentWindow: true
+    }).then((tabs) => {
+      for (var tab of tabs) {
+        if (!tab.active){
+          await browser.tabs.update(tabId, {
+              highlighted: true,
+              active: false,
+          });
+        }else{
+          await browser.tabs.update(tabId, {
+            highlighted: true,
+        });
+        }
+      }
+    });
+  }
+}
+
+async function highlightSelectedTabs(center, tabs, dir) {
+  if(dir === "left"){
+    const tabsLeft = tabs.filter(tab => {
+      return tab.index < center.index;
+    });
+    for (let i = 0; i < tabsLeft.length; i++) {
+      tabLeft = tabsLeft[tabsLeft.length - i - 1];
```

Comment:
Is there a reason you are going from the left over? It seems like this could just be `for (const tab of tabsLeft) { await browser.tabs.update(tab.id, {highlighted: true})`